dependencies/src/main/java/com/google/cloud/tools/opensource/dependencies/DependencyGraphBuilder.java

```diff
@@ -337,19 +337,20 @@ private static DependencyGraph levelOrder(
   }
 
   /**
-   * Returns true if {@code exceptionAndPath.getPath} does not contain {@code optional} dependency
-   * and the path does not contain {@code scope:provided} dependency.
+   * Returns true if {@code dependencyPath} does not contain {@code optional} dependency and the
+   * path does not contain {@code scope:provided} dependency.
    */
-  private static boolean isUnacceptableResolutionException(ExceptionAndPath exceptionAndPath) {
-    ImmutableList<DependencyNode> dependencyNodes = exceptionAndPath.getPath();
+  public static boolean unacceptableMissingDependency(List<DependencyNode> dependencyPath) {
     boolean hasOptionalParent =
-        dependencyNodes.stream().anyMatch(node -> node.getDependency().isOptional());
+        dependencyPath.stream()
+            .filter(node -> node.getDependency() != null) // Root node does not have dependency
+            .anyMatch(node -> node.getDependency().isOptional());
     if (!hasOptionalParent) {
       return true;
     }
     boolean hasProvidedParent =
```

Comment:
ditto. I don't see how the "parent" comes into play here.

Reply:
Removed `parent`.